graphql_compiler/compiler/emit_sql.py

```diff
@@ -521,7 +659,7 @@ def fold(self, fold_scope_location):
         ][full_edge_name]
 
         # 3. initialize fold object
-        self._current_fold = SQLFoldObject(outer_alias, join_descriptor)
+        self._current_fold = SQLFoldObject(self.dialect, outer_alias, join_descriptor)
```

Comment:
Why do we sometimes use `self.dialect` and sometimes use `self._dialect`? What is the purpose of making `self.dialect` a property?